### PR TITLE
refactor: better Result types

### PR DESCRIPTION
**Pull request by TBonnin**
- Making Result types returns an instance of the interface so we can use dot notation `res.isOk()/res.isErr()` instead of `isOk(res)/isErr(res)`, avoiding to import the functions every time.
- It also allows us to extends Ok and Err with functions like `.unwrap()` that is very useful in tests. I also needed a `.map()` function recently.
- Shortening `resultOk/resultErr` to `Ok/Err`. imho result prefix doesn't add much and since Result is now used across the codebase adding more context isn't adding much value.
- Renaming `ok.res` to `ok.value` (all the `res.res` were a bit weird) and `err.err` to `err.error`

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/utils/lib/result.ts [hunk 1]
```diff
@@ -1,26 +1,51 @@
-export interface ResultRes<T> {
-    ok: true;
-    res: T;
+export interface Left<T, E extends Error> {
```
Comment (TBonnin) [inline packages/utils/lib/result.ts:5]: sticking with Left/Right convention used by functional programming languages

**Review by khaliqgant — APPROVED** [review]

File: packages/persist/lib/controllers/persist.controller.ts [hunk 2]
```diff
@@ -204,33 +204,33 @@ class PersistController {
...
                 environment_id: environmentId,
                 activity_log_id: activityLogId,
-                content: `There was an issue with the batch ${persistType}. ${formatting.err.message}`,
+                content: `There was an issue with the batch ${persistType}. ${formatting.error.message}`,
                 timestamp: Date.now()
             });
-            await logCtx.error('There was an issue with the batch', { error: formatting.err, persistType });
-            const res = resultErr(`Failed to ${persistType} records ${activityLogId}`);
+            await logCtx.error('There was an issue with the batch', { error: formatting.error, persistType });
+            const err = Error(`Failed to ${persistType} records ${activityLogId}`);
```
Comment (bodinsamuel) [inline packages/persist/lib/controllers/persist.controller.ts]:
```suggestion
            const err = new Error(`Failed to ${persistType} records ${activityLogId}`);
```

File: packages/persist/lib/controllers/persist.controller.ts [hunk 3]
```diff
@@ -204,33 +204,33 @@ class PersistController {
...
-            span.setTag('error', res.err).finish();
-            return res;
+            span.setTag('error', err).finish();
+            return Err(err);
         }
 
         const syncConfig = await getSyncConfigByJobId(syncJobId);
         if (syncConfig && !syncConfig?.models.includes(model)) {
-            const res = resultErr(`The model '${model}' is not included in the declared sync models: ${syncConfig.models}.`);
+            const err = Error(`The model '${model}' is not included in the declared sync models: ${syncConfig.models}.`);
```
Comment (bodinsamuel) [inline packages/persist/lib/controllers/persist.controller.ts]:
```suggestion
            const err = new Error(`The model '${model}' is not included in the declared sync models: ${syncConfig.models}.`);
```

**Review by bodinsamuel — APPROVED** [review]
Seems convenient 👍🏻 
(still not a huge fan of an uppercase function name)

**Comment by TBonnin** [comment]
> Seems convenient 👍🏻 (still not a huge fan of an uppercase function name)

It is not a function, it's a constructor (sort of) 😉

**Pull request merged**